File: app/src/main/java/fup/prototype/robprototype/ProtoApplication.java
package fup.prototype.robprototype;

import android.app.Application;

import fup.prototype.robprototype.di.component.AppComponent;
import fup.prototype.robprototype.di.component.DaggerAppComponent;

public class ProtoApplication extends Application {

    private static AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        if (appComponent == null) {
            appComponent = DaggerAppComponent.builder().application(this).build();
        }
    }

    public static AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/java/com/rxdroid/app/ProtoApplication.kt
package com.rxdroid.app

import android.app.Application
import com.rxdroid.app.di.appModuleList
import org.koin.android.ext.android.startKoin
import timber.log.Timber

class ProtoApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        startKoin(this, appModuleList)
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
    }
}


